ppr-ui/src/views/discharge/ConfirmDischarge.vue

```diff
@@ -209,109 +205,36 @@ export default defineComponent({
+          localState.registrationNumber !== getRegistrationNumber.value) {
+        emit('error', 'Invalid Registration State')
         goToDash()
-        return
-      }
-      localState.financingStatementDate = new Date()
-      const financingStatement = await getFinancingStatement(
-        true,
-        localState.registrationNumber
-      )
-      if (financingStatement.error) {
-        localState.dataLoadError = true
-        emitError(financingStatement.error)
-      } else {
-        // set collateral summary
-        if (
-          financingStatement.generalCollateral &&
-          !financingStatement.vehicleCollateral
-        ) {
-          localState.collateralSummary = 'General Collateral and 0 Vehicles'
-        } else if (financingStatement.generalCollateral) {
-          localState.collateralSummary = `General Collateral and ${financingStatement.vehicleCollateral.length} ` +
-            'Vehicles'
-        } else if (financingStatement.vehicleCollateral) {
-          localState.collateralSummary = `No General Collateral and ${financingStatement.vehicleCollateral.length} ` +
-            'Vehicles'
-        } else {
-          localState.collateralSummary += 'No Collateral'
-        }
-        if (financingStatement.vehicleCollateral?.length === 1) {
-          localState.collateralSummary = localState.collateralSummary
-            .replace('Vehicles', 'Vehicle')
-        }
-        // load data into the store
-        const registrationType = AllRegistrationTypes.find((reg, index) => {
-          if (reg.registrationTypeAPI === financingStatement.type) {
-            return true
-          }
-        })
-        const parties = {
-          valid: true,
-          registeringParty: null,
-          securedParties: financingStatement.securedParties,
-          debtors: financingStatement.debtors
-        } as AddPartiesIF
-        setRegistrationCreationDate(financingStatement.createDateTime)
-        setRegistrationExpiryDate(financingStatement.expiryDate)
-        setRegistrationNumber(financingStatement.baseRegistrationNumber)
-        setRegistrationType(registrationType)
-        setAddSecuredPartiesAndDebtors(parties)
       }
     }
 
-    const onAppReady = async (val: boolean): Promise<void> => {
-      // do not proceed if app is not ready
-      if (!val) return
-      // redirect if not authenticated (safety check - should never happen) or if app is not open to user (ff)
-      if (!isAuthenticated.value || (!props.isJestRunning && !getFeatureFlag('ppr-ui-enabled'))) {
-        goToDash()
-        return
-      }
-
-      // get registration data from api and load into store
-      localState.submitting = true
-      await loadRegistration()
-      localState.submitting = false
-
-      // page is ready to view
-      emitHaveData(true)
-      localState.dataLoaded = true
-    }
+    /** Called when App is ready and this component can load its data. */
+    watch(() => props.appReady, (appReady: boolean) => {
```

Comment:
If we add immediate, to watcher I don't believe we need a watcher and onMount hook.